From c67a53577ca937fdad4cf359f04e89292aa5c8fe Mon Sep 17 00:00:00 2001
From: Artem Borisov <dedsa2002@gmail.com>
Date: Sat, 10 Nov 2018 17:19:17 +0000
Subject: [PATCH 2/5] Telephony: Don not call onUssdRelease for Huawei RIL

Huawei RIL doesn't seem to work properly with USSD_MODE_NW_RELEASE,
always releasing USSD when it should be finished instead.
Let's explicitly call onUssdFinished in this case.

Change-Id: I69faed1c51d4582834879975d6ab13daf7f48ad4

Conflicts:
	src/java/com/android/internal/telephony/GsmCdmaPhone.java
---
 src/java/com/android/internal/telephony/GsmCdmaPhone.java | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/java/com/android/internal/telephony/GsmCdmaPhone.java b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
index 1fa1621b6..a5ffb5da0 100644
--- a/src/java/com/android/internal/telephony/GsmCdmaPhone.java
+++ b/src/java/com/android/internal/telephony/GsmCdmaPhone.java
@@ -2154,6 +2154,8 @@ public class GsmCdmaPhone extends Phone {
             if (isUssdRelease) {
                 if (SystemProperties.getBoolean("ro.telephony.isHisiRIL", false)) {
                     found.onUssdFinished(ussdMessage, isUssdRequest);
+                } else if (SystemProperties.getBoolean("persist.sys.radio.huawei", false)) {
+                    found.onUssdFinished(ussdMessage, isUssdRequest);
                 } else {
                     found.onUssdRelease();
                 }
-- 
2.17.1

